src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,50 @@
 package core.basesyntax;
 
+import java.text.DateFormat;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.time.LocalDate;
+import java.time.ZoneId;
+import java.util.Date;
+import java.util.Locale;
+
 public class SalaryInfo {
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        StringBuilder res = new StringBuilder();
+        try {
+            SimpleDateFormat formatter = new SimpleDateFormat("dd.MM.yyyy");
```

Comment:
you should make pattern "dd.MM.yyyy" like a constant and "formatter" too and put them in private field of the class. It was written in check list if you read it attendently

Reply:
I did refactor according to remarks